src/bridge/handler.rs

```diff
@@ -14,12 +17,54 @@ use crate::{
     settings::SETTINGS,
 };
 
+struct NotificationEvent {
+    event_name: String,
+    arguments: Vec<Value>,
+}
+
 #[derive(Clone)]
-pub struct NeovimHandler {}
+pub struct NeovimHandler {
+    sender: Arc<Mutex<Sender<NotificationEvent>>>,
+}
 
 impl NeovimHandler {
     pub fn new() -> Self {
-        Self {}
+        let (sender, receiver): (Sender<NotificationEvent>, Receiver<NotificationEvent>) =
+            channel();
+        thread::spawn(|| {
+            for event in receiver {
```

Comment:
Just out of curiosity, have you tried just sending the events in the handler directly rather than sending over a channel? I am guessing there may be some perf issues if done that way, but its much simpler, so might be worth trying at the very least